src/librustc_mir/build/block.rs

```diff
@@ -120,10 +121,19 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                             opt_destruction_scope.map(|de|(de, source_info)), block, |this| {
                                 let scope = (init_scope, source_info);
                                 this.in_scope(scope, lint_level, block, |this| {
-                                    this.expr_into_pattern(block, pattern, init)
+                                    this.expr_into_pattern(block, ty, pattern, init)
                                 })
                             }));
                     } else {
+                        // FIXME: We currently only insert `UserAssertTy` statements for patterns
```

Comment:
Can we say `FIXME(#47184)`? We no longer mandate it, but I still prefer to have issue labels where we can.